File: agent-service/schemas/team.py
from typing import List, Literal, Optional
from pydantic import BaseModel
class TeamConfig(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    mode: Literal['sequential','round_robin','selector','manual'] = 'sequential'
    maxTurns: int = 20
    retryLimit: int = 2
    termination: str = 'workflow_completed'
    agents: List[str]

File: src/types/team.ts
export type TeamMode = 'sequential' | 'round_robin' | 'selector' | 'manual';

export interface TeamAgentConfig {
  agentId: string;
  name: string;
  role: string;
  order: number;
  enabled: boolean;
}

export interface TeamConfig {
  id: string;
  name: string;
  mode: TeamMode;
  maxTurns: number;
  retryLimit: number;
  termination: string;
  executionPolicy: string;
  modelOverride: string;
  default: boolean;
  enabled: boolean;
  agents: TeamAgentConfig[];
}
